consensus/safety-rules/src/process.rs

```diff
@@ -32,6 +34,7 @@ impl Process {
                 storage,
                 verify_vote_proposal_signature,
             }),
+            timeout: config.timeout,
```

Comment:
why not make this part of ProcessData too? This creates an interesting asymmetry.

Reply:
ProcessData is the result of invoking execute;  the invocation of execute relies on network_timeout to produce the ProcessData

Reply:
I don't follow, the exact same code path is used by data.server_addr. It does look pretty awkward for the data for NetworkService to come from two different sources.

Reply:
Ahh, now that I see the RemoteService(Config) in addition to the trait RemoteService, the comment is much more clear.

I agree with that awkward asymmetry, but it rears its head again when Thread invokes RemoteService::execute and does not have access to a RemoteService(Config).

The bigger issue blocking doing this in a clean fashion lies in RemoteService::execute being applied to structs which do not implement the RemoteService trait.

static functions considered evil. ;)